### PR TITLE
feat: Use last modified headers

### PR DESCRIPTION
**Pull request by JamieSinn**

### CODE REVIEW

**Review by jonathannorris — APPROVED** [review]

File: .github/workflows/unit-tests.yaml [hunk 1]
```diff
@@ -15,12 +15,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby-version: [ '3.0', '3.0.4', '3.1.0', '3.2.0' ]
+        ruby-version: [ '3.1.0', '3.2.0', '3.3.0' ]
```
Comment (jonathannorris) [inline .github/workflows/unit-tests.yaml:18]: do we need to remove `3.0` support?

File: devcycle-ruby-server-sdk.gemspec [hunk 2]
```diff
@@ -21,12 +21,12 @@ Gem::Specification.new do |s|
   s.summary     = "DevCycle Bucketing API Ruby Gem"
   s.description = "DevCycle Ruby Server SDK, for interacting with feature flags created with the DevCycle platform."
   s.license     = "MIT"
-  s.required_ruby_version = ">= 2.4"
+  s.required_ruby_version = ">= 3.1"
```
Comment (jonathannorris) [inline devcycle-ruby-server-sdk.gemspec:24]: So are we going to need to mark this as a major version if we are going from 2.4 -> 3.1?

Comment (JamieSinn) [reply devcycle-ruby-server-sdk.gemspec:24]: We can - but 3.0 is EOL https://endoflife.date/ruby
https://www.ruby-lang.org/en/downloads/

File: .github/workflows/unit-tests.yaml (hunk 1 above)
Comment (JamieSinn) [reply .github/workflows/unit-tests.yaml:18]: 3.0 is EOL https://endoflife.date/ruby

https://www.ruby-lang.org/en/downloads/

File: lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb [hunk 3]
```diff
@@ -53,22 +55,43 @@ def fetch_config
...
         when 200
-          @logger.debug("New config received, etag: #{resp.headers['Etag']}")
-          set_config(resp.body, resp.headers['Etag'])
-          @logger.debug("New config stored, etag: #{@config_e_tag}")
+          @logger.debug("New config received, etag: #{resp.headers['Etag']} LM:#{resp.headers['Last-Modified']}")
+          lm_header = resp.headers['Last-Modified']
+          begin
+            lm_timestamp = Time.rfc2822(lm_header)
+            current_lm = Time.rfc2822(@config_last_modified)
+            if lm_timestamp == "" && @config_last_modified == "" || (current_lm.utc < lm_timestamp.utc)
```
Comment (jonathannorris) [inline lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb:85]: are there any null / nils in ruby?

File: .github/workflows/unit-tests.yaml (hunk 1 above)
Comment (jonathannorris) [reply .github/workflows/unit-tests.yaml:18]: only released 3.5 years ago... I assume a lot of folks are still using it.

File: lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb (hunk 3 above)
Comment (JamieSinn) [reply lib/devcycle-ruby-server-sdk/localbucketing/config_manager.rb:85]: https://blog.arkency.com/2017/07/nil-empty-blank-ruby-rails-difference/

File: .github/workflows/unit-tests.yaml (hunk 1 above)
Comment (JamieSinn) [reply .github/workflows/unit-tests.yaml:18]: We don't support EOL versions in any other SDK.

File: devcycle-ruby-server-sdk.gemspec (hunk 2 above)
Comment (jonathannorris) [reply devcycle-ruby-server-sdk.gemspec:24]: yea like if we make this big jump in supported versions we probably need to make it a major version.

Comment (JamieSinn) [reply devcycle-ruby-server-sdk.gemspec:24]: kk - i'll mark major when we release

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**